### PR TITLE
Get collections must have only value and nextLink (RPC-Get-v1-09, RPC-Arg-V1-01)

### PR DESCRIPTION
**Pull request by bdefoy**
RPC-Get-v1-09: 
> LIST calls MUST have **ONLY** "value" and "nextLink" properties in its model.

RPC-Arg-V1-01:L
> Do not model embedded resource or reference child resource qualifications on parent resources for LIST calls. This causes inconsistent query results between portal and ARG.

### CODE REVIEW

**Comment by konrad-jamrozik** [comment]
I would appreciate a PR description explaining this change, including what kind of testing has been conducted to confirm this change is correct.

File: packages/rulesets/src/spectral/az-arm.ts [hunk 1]
```diff
@@ -211,6 +212,18 @@ const ruleset: any = {
         function: falsy,
       },
     },
+    // RPC Codes: RPC-Get-V1-09, RPC-Arg-V1-01, RPC-Get-V1-06
```
Comment (konrad-jamrozik) [inline packages/rulesets/src/spectral/az-arm.ts:215]: Can we link here to a doc elaborating on what are these codes?

Comment (bdefoy) [reply packages/rulesets/src/spectral/az-arm.ts:215]: We do not have a place where these are published yet. The descriptions are currently only available in the ARM OneNote.

Comment (konrad-jamrozik) [reply packages/rulesets/src/spectral/az-arm.ts:258]: @bdefoy could you link me to the relevant section in the ARM OneNote? I think we should clarify in the code comments and docs the canonical source of this rules is not public, and thus what is the process of cross-checking/verifying correctness of them. E.g. who to contact if there are any questions about this rule.

**Comment by konrad-jamrozik** [comment]
The PR description mentions code `RPC-Get-v1-09` and `RPC-Arg-V1-01` but the added code comment also mentions `RPC-Get-V1-06`. What is this extra code about?

Also, could you help me understand how the changes made in this PR fulfill the rule `RPC-Arg-V1-01` ? I don't see it based on the code and doc changes and the PR description.

File: packages/rulesets/src/spectral/az-arm.ts (hunk 1 above)
Comment (rkmanda) [reply packages/rulesets/src/spectral/az-arm.ts:258]: For all ARM specific rules, you need to contact ARM RPC Governance Team <armrpcgovernanceteam@microsoft.com>

**Review by rkmanda — APPROVED** [review]
:shipit:

**Pull request merged**